readthedocs/projects/tasks.py

```diff
@@ -1146,6 +1150,56 @@ def fileify(version_pk, commit):
         )
 
 
+def _update_intersphinx_data(version, path, commit):
+    """
+    Update intersphinx data for this version
+
+    :param version: Version instance
+    :param path: Path to search
+    :param commit: Commit that updated path
+    """
+    object_file = os.path.join(path, 'objects.inv')
+
+    # These classes are from copied from Sphinx tests
+    class MockConfig:
+        intersphinx_timeout = None
+        tls_verify = False
+
+    class MockApp:
+        srcdir = ''
+        config = MockConfig()
+
+        def warn(self, msg):
+            log.warning('Sphinx MockApp: %s', msg)
+
+    invdata = intersphinx.fetch_inventory(MockApp(), '', object_file)
+    for key in sorted(invdata or {}):
+        domain, _type = key.split(':')
+        for name, einfo in sorted(invdata[key].items()):
+            url = einfo[2]
```

Comment:
I'd add a small comment with previous to this line, with the expected structure of this `einfo` object. Like

    # (key, name, url, doc type, etc)
    # ('py', 'function', '/some-nice-url/python-function/', 'func', ...)

or... a link to the docs.

I found debugging this complicated later.